### PR TITLE
fix(indexer): cleanup custom client code with op-service changes

### PR DESCRIPTION
**Pull request by hamdiallam**
- Remove custom client code & c/p rpc metrics in favor of op-service
- move the golang api client into `/api` for clarity. `/client` is a bit confusing as a general client

### CODE REVIEW

**Comment by hamdiallam** [comment]
* **#10546** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/10546?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#10545** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/10545?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @hamdiallam and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: indexer/node/header_traversal_test.go [hunk 1]
```diff
@@ -55,25 +55,31 @@ func TestHeaderTraversalNextHeadersNoOp(t *testing.T) {
 }
 
 func TestHeaderTraversalNextHeadersCursored(t *testing.T) {
-	client := new(MockEthClient)
+	client := &testutils.MockClient{}
+	rpc := &testutils.MockRPC{}
+	client.Mock.On("RPC").Return(rpc)
```
Comment (protolambda) [inline indexer/node/header_traversal_test.go]: I think the mock-clients may need an `m.AssertExpectations(t)` at the end of the test.

File: indexer/node/header_traversal_test.go [hunk 2]
```diff
@@ -82,17 +88,22 @@ func TestHeaderTraversalNextHeadersCursored(t *testing.T) {
 }
 
 func TestHeaderTraversalNextHeadersMaxSize(t *testing.T) {
-	client := new(MockEthClient)
+	client := &testutils.MockClient{}
+	rpc := &testutils.MockRPC{}
+	client.Mock.On("RPC").Return(rpc)
```
Comment (protolambda) [inline indexer/node/header_traversal_test.go]: Same here

File: indexer/node/header_traversal_test.go [hunk 3]
```diff
@@ -112,24 +129,40 @@ func TestHeaderTraversalNextHeadersMaxSize(t *testing.T) {
 }
 
 func TestHeaderTraversalMismatchedProviderStateError(t *testing.T) {
-	client := new(MockEthClient)
+	client := &testutils.MockClient{}
+	rpc := &testutils.MockRPC{}
+	client.Mock.On("RPC").Return(rpc)
```
Comment (protolambda) [inline indexer/node/header_traversal_test.go]: Same here

**Review by protolambda — APPROVED** [review]
Looks good, just a few tests that use mocks can use an expectations-check

File: indexer/node/header_traversal_test.go (hunk 1 above)
Comment (hamdiallam) [reply indexer/node/header_traversal_test.go]: good call, wasn't aware this needed to be called, updated!

**Pull request merged**